macros/src/parser/lifetimes.rs

```diff
@@ -0,0 +1,98 @@
+use quote::quote;
+use std::ops::Deref;
+use syn::{parse, spanned::Spanned, GenericArgument, Lifetime, PathArguments, Type};
+
+#[derive(Debug, Clone)]
+pub struct Lifetimes(Vec<Lifetime>);
```

Comment:
Can we make this a named struct instead of a tuple? The .0.<xxx> syntax is really opaque in my experience and makes it difficult to add functionality later on.